lib/matplotlib/quiver.py

```diff
@@ -779,142 +779,138 @@ def _h_arrows(self, length):
+    Whether the empty barbs (circles) that are drawn should be filled with
+    the flag color.  If they are not filled, the center is transparent.
 
-  *rounding*:
-    A flag to indicate whether the vector magnitude should be rounded
-    when allocating barb components.  If True, the magnitude is
-    rounded to the nearest multiple of the half-barb increment.  If
-    False, the magnitude is simply truncated to the next lowest
-    multiple.  Default is True
+rounding : bool, default: True
+    Whether the vector magnitude should be rounded when allocating barb
+    components.  If True, the magnitude is rounded to the nearest multiple
+    of the half-barb increment.  If False, the magnitude is simply truncated
+    to the next lowest multiple.
 
-  *barb_increments*:
+barb_increments : dict, optional
     A dictionary of increments specifying values to associate with
     different parts of the barb. Only those values one wishes to
     override need to be included.
 
-        - 'half' - half barbs (Default is 5)
-
-        - 'full' - full barbs (Default is 10)
-
-        - 'flag' - flags (default is 50)
-
-  *flip_barb*:
-    Either a single boolean flag or an array of booleans.  Single
-    boolean indicates whether the lines and flags should point
-    opposite to normal for all barbs.  An array (which should be the
-    same size as the other data arrays) indicates whether to flip for
-    each individual barb.  Normal behavior is for the barbs and lines
-    to point right (comes from wind barbs having these features point
-    towards low pressure in the Northern Hemisphere.)  Default is
-    False
-
-Barbs are traditionally used in meteorology as a way to plot the speed
-and direction of wind observations, but can technically be used to
-plot any two dimensional vector quantity.  As opposed to arrows, which
-give vector magnitude by the length of the arrow, the barbs give more
-quantitative information about the vector magnitude by putting slanted
-lines or a triangle for various increments in magnitude, as show
-schematically below::
-
- :     /\    \\
- :    /  \    \\
- :   /    \    \    \\
- :  /      \    \    \\
- : ------------------------------
+    - 'half' - half barbs (Default is 5)
+    - 'full' - full barbs (Default is 10)
+    - 'flag' - flags (default is 50)
 
-.. note the double \\ at the end of each line to make the figure
-.. render correctly
-
-The largest increment is given by a triangle (or "flag"). After those
-come full lines (barbs). The smallest increment is a half line.  There
-is only, of course, ever at most 1 half line.  If the magnitude is
-small and only needs a single half-line and no full lines or
-triangles, the half-line is offset from the end of the barb so that it
-can be easily distinguished from barbs with a single full line.  The
-magnitude for the barb shown above would nominally be 65, using the
-standard increments of 50, 10, and 5.
+flip_barb : bool, default: False
```

Comment:
Please leave the part about taking an array of bools—that’s an intended feature. While it doesn’t work right now I’m putting in a PR to make it work as the very next thing I code.

Reply:
Since I would like to target this doc change to 3.1.1, I'd prefer to describe the behavior as is.I assume that the array handling will go into 3.2.

Additionally, IMHO the section should be rewritten anyway; and I'm not sure about the implementation details (in particular accepting 1D or 2D).

You can easily add the docs to your PR. It's really not much more than

~~~
flip_barb : bool or array-like of bool, default: False
    [existing doc]
    A single value applies to all barbs. Individual barbs can be flipped by passing
    a bool array of the same size as *U* and *V*.
~~~

Reply:
Well, I want the fix for the arrays to go in 3.1.1--because right now it'd be a bugfix for documented behavior. I don't want to wait for 3.2 for it as a feature.